packages/wallet/src/redux/protocols/advance-channel/reducer.ts

```diff
@@ -147,57 +145,175 @@ function initializeWithNewChannel(
       sharedData,
     };
   } else {
-    throw new Error('Unimplemented');
+    const protocolState = states.channelUnknown({
+      ...initializeChannelArgs,
+      processId,
+    });
+
+    return { protocolState, sharedData };
   }
 }
 
-function initializeWithExistingChannel(channel, processId, sharedData) {
-  const { ourIndex, channelId } = channel;
-  return { protocolState: states.notSafeToSend({ processId, channelId, ourIndex }), sharedData };
+const channelUnknownReducer: ProtocolReducer<states.AdvanceChannelState> = (
+  protocolState: states.ChannelUnknown,
+  sharedData,
+  action: CommitmentsReceived,
+) => {
+  const { ourIndex, privateKey, commitmentType } = protocolState;
+  const channelId = getChannelId(action.signedCommitments[0].commitment);
+
+  const checkResult = checkAndInitialize(sharedData, action.signedCommitments[0], privateKey);
+  if (!checkResult.isSuccess) {
```

Comment:
Will the stack trace be easier to interpret if the error is thrown next to the logic where the failure occurs?